Docker (#2400)
SumoLogic/sumologic-documentation: docs/integrations/containers-orchestration/opentelemetry/docker-opentelemetry.md

---
id: docker-opentelemetry
title: Docker - OpenTelemetry Collector
sidebar_label: Docker - OTel Collector
description: Learn about the Sumo Logic OpenTelemetry app for Docker.
---

import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/integrations/containers-orchestration/docker-icon.png')} alt="Thumbnail icon" width="120"/>

The Sumo Logic app for Docker ULM is a unified logs and metrics app that enables you to monitor Docker deployment. The app provides preconfigured dashboards that include information about container state and resource usage, including information on CPU, memory, block I/O, and network.

> OpenTelemetry collector runs on the same host as Docker and helps to collect Docker metric and container event logs. 

Reviewer: ```suggestion
The Sumo Logic OpenTelemetry collector will run on the same host as Docker and help to collect Docker metric and container event logs. 
```

:::info
The Sumo Logic app for Docker supports Docker version 23.0.2.
:::

<img src={useBaseUrl('img/integrations/containers-orchestration/Docker-Schematics.png')} alt="Docker-Schematics"/>

## Fields creation in Sumo Logic for Docker

Following are the tags which will be created as part of the Docker app installation, if not already present.

* `sumo.datasource`. Has a fixed value of **docker**.

## Prerequisites

This section provides instructions for configuring metrics and log collection for the Sumo Logic app for Docker.

### Metric collection

Metrics are collected through the [Docker Stats Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/README.md) of OpenTelemetry.

### Log collection

To collect the Docker container event logs, the following command needs to be executed on the host machine and needs to be kept running, for monitoring all the Docker container-related events. The following command also needs a JSON file path where these container events can be dumped.

```
docker events docker events -f 'type=container' --format '{{json .}}' > <PATH_TO_JSON> & disown
```

> Path to this JSON file will be required in the [next step](#step-2-configure-integration), where events are sent to Sumo Logic through a filelog receiver and seen as part of the **Docker - Overview** dashboard. Also, you can add additional parameters to this command to send events for specific containers. Learn [more](https://docs.docker.com/engine/reference/commandline/events/). 

Reviewer: ```suggestion
Path to this JSON file will be required in the [next step](#step-2-configure-integration), where events are sent to Sumo Logic through a filelog receiver and seen as part of the **Docker - Overview** dashboard. Also, you can add additional parameters to this command to send events for specific containers. [Learn more](https://docs.docker.com/engine/reference/commandline/events/). 
```

## Collecting Logs, Metrics, and Installing app for Docker​

> Follow the below steps for collecting logs, metric and installing the app:

Reviewer: ```suggestion
Follow the below steps for collecting logs, metrics, and installing the app.
```

- **Docker Event log location**. Enter the path of the JSON file generated through the command in the prerequisite section.
- **Excluded Image List**. A list of strings, [regexes](https://golang.org/pkg/regexp/), or [globs](https://github.com/gobwas/glob) whose referent container image names will not be among the queried containers for scrapping metrics. Learn more about [*excluded_images*](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/README.md#configuration).

You can add any custom fields which you want to tag along with the data ingested in sumo.

Click on the **Download YAML File** button to get the yaml file.

<img src={useBaseUrl('img/integrations/containers-orchestration/Docker-YAML.png')} alt="Docker-YAML"/>

### Step 3: Send logs to Sumo

Once you have downloaded the yaml file as described in the previous step, follow the below steps based on your platform.

1. Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Docker instance that needs to be monitored.
1. Place `Env` file in the `/etc/otelcol-sumo/env/` directory.
1. Restart the collector using:
  ```
  sudo systemctl restart otelcol-sumo
  ```

> After successfully executing the above command, Sumo Logic will start receiving data from your host machine.
> 
> Click **Next**. This will install the app (dashboards and monitors) to your Sumo Logic Org.
> 
> Dashboard panels will start to fill automatically. It's important to note that each panel fills with data matching the time range query and received since the panel was created. Results won't immediately be available, but within 20 minutes, you'll see full graphs and maps.

Reviewer: ```suggestion
{@import ../../../reuse/apps/opentelemetry/send-logs-outro.md}
```

- **Number of Bytes Transferred to/from the Disk**. Number of bytes transferred to and from disk by each container

### Docker - Network Usage

<img src={useBaseUrl('img/integrations/containers-orchestration/Docker-Network-Usage-Otel.png')} alt="Docker-Network-Usage"/>

- **Average Rx Bytes by Container**. Displays the average number of bytes received per timeslice by each container.
- **Average Tx Bytes by Container**. Displays the average number of bytes transmitted per timeslice by each container.
- **Count of Rx Dropped Packets**. Count of received packets dropped by each container per timeslice.
- **Count of Rx Error Packets**. Count of error packets received per timeslice by each container.
- **Average Rx Packets by Container**. Average packets received per timeslice by each container.
- **Average Tx Packets by Container**. Average packets transmitted per timeslice by each container.
- **Count of Tx Dropped Packets**. Count of packets dropped during transmission per timeslice by each container.
- **Count of Tx Error Packets**. Count of error packets transmitted per timeslice by each container.

> 
> 
> 
> 
> 

Reviewer: ```suggestion
```
